test/system/030-run.bats

```diff
@@ -901,4 +901,24 @@ $IMAGE--c_ok" \
     run_podman rm $ctr_name
 }
 
+@test "podman run --privileged as root with systemd will not mount /dev/tty" {
+    skip_if_rootless "this test only makes sense as root"
+
+    ctr_name="container-$(random_string 5)"
+    run_podman run --rm -d --privileged --systemd=always --name "$ctr_name" "$IMAGE" /home/podman/pause
+
+    TTYs=$(ls /dev/tty*|sed '/^\/dev\/tty$/d')
+
+    if [[ $TTYs = "" ]]; then
+        false
+    else
+        run_podman exec "$ctr_name" ls /dev/
+        run_output=$output
+        [[ $(echo "$run_output" | grep -c "tty") = "1"  ]]
+        [[ $(echo "$run_output" | grep "tty") = "tty"  ]]
```

Comment:
Courtesy alert: again, this is regular-BATS stuff, which is impossible for anyone to debug on failure. In our environment, one correct way to do that is:
```bash
   assert "$(grep tty <<<$output)" = "tty" "No /dev/ttyN devices in container"
```
(no need to define `run_output`, it's not a useful variable). (And no need for the `-c` test either)

Reason: if this test fails, `assert` will give an error of the form "this is what I expected, and this is what I got". With regular BATS, you just get "meh, test failed but I'm not going to give you any useful information about why".

See  https://github.com/containers/podman/blob/main/test/system/README.md